sql/hive/src/test/scala/org/apache/spark/sql/hive/orc/HiveOrcSourceSuite.scala

```diff
@@ -190,4 +192,103 @@ class HiveOrcSourceSuite extends OrcSuite with TestHiveSingleton {
+            // Hive throws an exception if I assign the location in the create table statement.
+            hiveClient.runSqlHive(
+              s"ALTER TABLE hive_orc SET LOCATION '$dataDir'")
+            hiveClient.runSqlHive(
+              """
+                |INSERT INTO TABLE hive_orc
+                |VALUES (1, 1, 'orc1'), (2, 2, 'orc2')""".stripMargin)
+
+            withTable("tbl1", "tbl2", "tbl3", "tbl4") {
+              val parentDirStatement =
+                s"""
+                   |CREATE EXTERNAL TABLE tbl1(
+                   |  c1 int,
+                   |  c2 int,
+                   |  c3 string)
+                   |STORED AS orc
+                   |LOCATION '${parentDir}'""".stripMargin
+              sql(parentDirStatement)
+              val parentDirSqlStatement = s"select * from tbl1"
+              if (convertMetastore) {
+                checkAnswer(sql(parentDirSqlStatement), Nil)
+              } else {
+                checkAnswer(sql(parentDirSqlStatement),
+                  (1 to 2).map(i => Row(i, i, s"orc$i")))
+              }
+
+              val l3DirStatement =
+                s"""
+                   |CREATE EXTERNAL TABLE tbl2(
+                   |  c1 int,
+                   |  c2 int,
+                   |  c3 string)
+                   |STORED AS orc
+                   |LOCATION '${l3Dir}'""".stripMargin
+              sql(l3DirStatement)
+              val l3DirSqlStatement = s"select * from tbl2"
+              if (convertMetastore) {
+                checkAnswer(sql(l3DirSqlStatement), Nil)
+              } else {
+                checkAnswer(sql(l3DirSqlStatement),
+                  (1 to 2).map(i => Row(i, i, s"orc$i")))
+              }
+
+              val wildcardStatement =
+                s"""
+                   |CREATE EXTERNAL TABLE tbl3(
+                   |  c1 int,
+                   |  c2 int,
+                   |  c3 string)
+                   |STORED AS orc
+                   |LOCATION '$wildcardParentDir'""".stripMargin
+              sql(wildcardStatement)
+              val wildcardSqlStatement = s"select * from tbl3"
+              if (convertMetastore) {
+                checkAnswer(sql(wildcardSqlStatement),
+                  (1 to 2).map(i => Row(i, i, s"orc$i")))
+              } else {
+                checkAnswer(sql(wildcardSqlStatement), Nil)
+              }
+
+              val wildcardL3Statement =
+                s"""
+                   |CREATE EXTERNAL TABLE tbl4(
+                   |  c1 int,
+                   |  c2 int,
+                   |  c3 string)
+                   |STORED AS orc
+                   |LOCATION '$wildcardL3Dir'""".stripMargin
+              sql(wildcardL3Statement)
+              val wildcardL3SqlStatement = s"select * from tbl4"
+              checkAnswer(sql(wildcardL3SqlStatement), Nil)
+            }
+          } finally {
+            hiveClient.runSqlHive("DROP TABLE IF EXISTS hive_orc")
```

Comment:
It seems that we need to clean up `tbl1` ~ `tbl4`, too.

Reply:
@dongjoon-hyun at line 221, I put the `tbl1` ~ `tbl4` with the `withTable`, I think it will get dropped. I tried to run it couple time in intellij, it seems work fine. what do you think?

Reply:
Got it. I missed that.